microsoft/testsuites/dpdk/dpdktestpmd.py

```diff
@@ -111,45 +115,69 @@ def _install(self) -> bool:
         self.dpdk_path = self.node.working_path.joinpath(self._dpdk_repo_path_name)
         if result.exit_code == 0:  # tools are already installed
             return True
-        if self._install_dependencies():
-            node = self.node
-            git_tool = node.tools[Git]
-            echo_tool = node.tools[Echo]
+        self._install_dependencies()
+        node = self.node
+        git_tool = node.tools[Git]
+        echo_tool = node.tools[Echo]
+        wget_tool = node.tools[Wget]
+        tar_tool = node.tools[Tar]
+
+        if self._dpdk_source and self._dpdk_source.endswith(".tar.gz"):
+            if self._dpdk_branch:
+                node.log.warn(
+                    (
+                        "DPDK tarball source does not need dpdk_branch defined. "
+                        "User-defined variable dpdk_branch will be ignored."
+                    )
+                )
+            working_path = node.working_path.as_posix()
+            wget_tool.get(
+                self._dpdk_source,
+                working_path,
+            )
+            dpdk_filename = self._dpdk_source.split("/")[-1]
+            node.log.info(node.execute("ls -la", cwd=node.working_path))
+            dpdk_foldername = tar_tool.extract(
```

Comment:
How about to support dest path in tar? So it doesn't need extra mv command.

Reply:
I will do this, but let me do it in a second PR. If I'm going to add that I'd like to change this tar tool to use the pathlib and that seems like it deserves it's own commit.

Reply:
Can I change dest_dir to use pathlib? Otherwise appending the new output folder onto the dest_dir gets weird

Reply:
Going to raise a separate PR to enable renaming the output dir and we can hold this one until it's finished

Reply:
Ah this is actually more complicated than I thought at first. A tar can make lots of output folders from one file

Reply:
It should be able to rename the root path to extract. It should be easy. If the folder is in the tar file, it's ok to live with it, we don't need to rename it.

Reply:
It's not as straightforward as you would think, I opened another PR for this though so we can discuss further there